<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>867</width>
    <height>551</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>queens problem</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QVBoxLayout" name="verticalLayout"/>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>867</width>
     <height>22</height>
    </rect>
   </property>
   <widget class="QMenu" name="menufile">
    <property name="title">
     <string>file</string>
    </property>
    <addaction name="actionsave"/>
    <addaction name="separator"/>
    <addaction name="actionabout"/>
   </widget>
   <widget class="QMenu" name="menuedit">
    <property name="title">
     <string>edit</string>
    </property>
    <addaction name="actioncopy_all_solutions"/>
    <addaction name="actioncopy_selected_solutions"/>
    <addaction name="actioncopy_result_message"/>
    <addaction name="separator"/>
    <addaction name="actioncopy_image"/>
   </widget>
   <widget class="QMenu" name="menuscan">
    <property name="title">
     <string>scan</string>
    </property>
    <addaction name="actionthread"/>
    <addaction name="separator"/>
    <addaction name="actiontransformations"/>
   </widget>
   <addaction name="menufile"/>
   <addaction name="menuedit"/>
   <addaction name="menuscan"/>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
   <addaction name="actionthread"/>
   <addaction name="separator"/>
   <addaction name="actiontransformations"/>
   <addaction name="separator"/>
   <addaction name="actionsave"/>
  </widget>
  <widget class="QStatusBar" name="statusBar">
   <property name="font">
    <font>
     <family>Courier New</family>
     <pointsize>10</pointsize>
    </font>
   </property>
  </widget>
  <widget class="QDockWidget" name="dockWidget">
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>1</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents">
    <layout class="QVBoxLayout" name="verticalLayout_3">
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout">
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QSpinBox" name="nQueens">
         <property name="prefix">
          <string>#queens </string>
         </property>
         <property name="minimum">
          <number>4</number>
         </property>
         <property name="maximum">
          <number>200</number>
         </property>
         <property name="value">
          <number>8</number>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="stopSols">
         <property name="suffix">
          <string>] solutions or more</string>
         </property>
         <property name="prefix">
          <string>stop when found [ </string>
         </property>
         <property name="maximum">
          <number>100000</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="horizontalSpacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_2">
       <property name="bottomMargin">
        <number>0</number>
       </property>
      </layout>
     </item>
     <item>
      <widget class="TabSolutions" name="tabSolutions">
       <property name="font">
        <font>
         <pointsize>11</pointsize>
        </font>
       </property>
       <property name="editTriggers">
        <set>QAbstractItemView::NoEditTriggers</set>
       </property>
      </widget>
     </item>
    </layout>
   </widget>
  </widget>
  <widget class="QDockWidget" name="dockWidget_2">
   <property name="features">
    <set>QDockWidget::NoDockWidgetFeatures</set>
   </property>
   <attribute name="dockWidgetArea">
    <number>2</number>
   </attribute>
   <widget class="QWidget" name="dockWidgetContents_2">
    <layout class="QVBoxLayout" name="verticalLayout_2" stretch="1">
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_4">
       <item>
        <widget class="ChessBoard" name="chessBoard"/>
       </item>
      </layout>
     </item>
    </layout>
   </widget>
  </widget>
  <action name="actionthread">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/VSlibQt/rsc/icons/tree.png</normaloff>
     <normalon>:/VSlibQt/rsc/icons/stop.png</normalon>:/VSlibQt/rsc/icons/tree.png</iconset>
   </property>
   <property name="text">
    <string>scan Tree &lt; (n!)</string>
   </property>
   <property name="toolTip">
    <string>scan tree</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+T</string>
   </property>
  </action>
  <action name="actionsave">
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/VSlibQt/rsc/icons/save.png</normaloff>:/VSlibQt/rsc/icons/save.png</iconset>
   </property>
   <property name="text">
    <string>save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actiontransformations">
   <property name="icon">
    <iconset resource="rsc.qrc">
     <normaloff>:/VSlibQt/rsc/icons/transformation.png</normaloff>:/VSlibQt/rsc/icons/transformation.png</iconset>
   </property>
   <property name="text">
    <string>transformations</string>
   </property>
   <property name="toolTip">
    <string>apply transformations (tH,tV,R90,mH,mV) to selected solutions</string>
   </property>
  </action>
  <action name="actionabout">
   <property name="text">
    <string>about</string>
   </property>
  </action>
  <action name="actioncopy_selected_solutions">
   <property name="text">
    <string>copy selected solutions</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+Shift+C</string>
   </property>
  </action>
  <action name="actioncopy_image">
   <property name="text">
    <string>copy board image</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+I</string>
   </property>
  </action>
  <action name="actioncopy_all_solutions">
   <property name="text">
    <string>copy all solutions</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+C</string>
   </property>
  </action>
  <action name="actioncopy_result_message">
   <property name="text">
    <string>copy result message</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>TabSolutions</class>
   <extends>QTableView</extends>
   <header location="global">tabsolutions.h</header>
  </customwidget>
  <customwidget>
   <class>ChessBoard</class>
   <extends>QOpenGLWidget</extends>
   <header location="global">chessboard.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="rsc.qrc"/>
 </resources>
 <connections/>
</ui>
